pkg/proxy/proxy.go

```diff
@@ -18,120 +18,137 @@ package proxy
 
 import (
 	"bytes"
+	"context"
+	"fmt"
 	"net/http"
 	"net/textproto"
 	"text/template"
 
 	"github.com/brancz/kube-rbac-proxy/pkg/authn"
 	"github.com/brancz/kube-rbac-proxy/pkg/authz"
-	"k8s.io/apiserver/pkg/authentication/user"
+
 	"k8s.io/apiserver/pkg/authorization/authorizer"
-	"k8s.io/klog/v2"
+	"k8s.io/apiserver/pkg/endpoints/request"
 )
 
+var _ authorizer.Authorizer = &krpAuthorizer{}
+
+const kubeRBACProxyParamsKey = iota
```

Comment:
> Is this some kind of context type to avoid collision in context.Context?
Yes, that's exactly it. It's a pattern used throughout kube, from what I was able to see.